### PR TITLE
Refactor out DEFINE_GAME_ACTION

### PR DESCRIPTION
**Pull request by ZehMatt**
The registration is also simplified, now Intellisense is actually capable of exploring those classes.

### CODE REVIEW

File: src/openrct2/actions/GameActionRegistry.cpp [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+            case GameCommand::PlaceFootpathAddition:
+                return "PlaceFootpathAddition";
+            case GameCommand::RemoveFootpathAddition:
+                return "RemoveFootpathAddition";
+            case GameCommand::GuestSetFlags:
+                return "GuestSetFlags";
+            case GameCommand::SetDate:
+                return "SetDate";
+            case GameCommand::Custom:
+                return "Custom";
```
Comment (Broxzier) [inline src/openrct2/actions/GameActionRegistry.cpp:360]: Shouldn't this be wrapped in `#ifdef ENABLE_SCRIPTING`?

May be nicer to refactor this entire function out, and have the names be part of the action classes, perhaps supplied where registered or as a template parameter. Probably better for another PR, though.

Comment (ZehMatt) [reply src/openrct2/actions/GameActionRegistry.cpp:360]: The id exists regardless, so I dunno.

File: src/openrct2/actions/GameActionRegistry.cpp [hunk 2]
```diff
@@ -0,0 +1,248 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2020 OpenRCT2 developers
```
Comment (Broxzier) [inline src/openrct2/actions/GameActionRegistry.cpp:2]:
```suggestion
 * Copyright (c) 2014-2021 OpenRCT2 developers
```

**Review by Broxzier — APPROVED** [review]

**Comment by Broxzier** [comment]
All changes look good, nice to see the macro gone. Haven't tested the build though.

**Pull request merged**